primo/utils/solvers.py

```diff
@@ -81,24 +82,20 @@ def get_solver(
         sol_obj.config.mip_gap = mip_gap
         sol_obj.config.time_limit = time_limit
 
-        for k, v in solver_options:
+        for k, v in solver_options.items():
             if hasattr(sol_obj.config, k):
                 setattr(sol_obj.config, k, v)
 
         return sol_obj
 
-    elif solver == "gurobi":
-        sol_obj = SolverFactory("gurobi", solver_io="python")
+    elif solver in ("gurobi", "gurobi_persistent"):
```

Comment:
The behavior of persistent solvers is different. That's why I did not include `gurobi_persistent` before. For example, I can do this with Gurobi, scip, glpk, and highs
```python
solver = get_solver("gurobi")
solver.solve(model)
```
But the above code does not work for `gurobi_persistent`. For persistent solvers, we need to set/add the instance manually.
```python
solver = get_solver("gurobi_persistent")
solver.set_instance(model)
solve.solve()
```